pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -563,6 +592,13 @@ func (gce *GCECloud) targetPoolURL(name, region string) string {
 	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/targetPools/%s", gce.projectID, region, name)
 }
 
+func (gce *GCECloud) getLoadBalancerName(service *v1.Service) string {
+        //GCE requires that the name of a load balancer starts with a lower case letter.
```

Comment:
I _think_ that is OK for GCE, where the limit is 63 chars.  We need enough chars from the UID to guarantee uniqueness, even if we have non-unique `namespace+'-'+name` (e.g. `"a-b" + "-" + "c"` vs `"a" + "-" + "b-c"`).

Also need to double-check  which characters are and are not allowed.

This is going to be trickier on AWS but I guess that the advantage of splitting up the logic here :-)